sdcm/nemesis.py

```diff
@@ -4417,6 +4423,106 @@ def disrupt_add_remove_mv(self):
         finally:
             self.unset_current_running_nemesis(cql_query_executor_node)
 
+    def disrupt_toggle_audit(self, store: AuditStore):
+        """
+            Enable audit log with all categories and user keyspaces (if audit already enabled, disable it and finish the Nemesis),
+            verify audit log content,
+            reduce categories by excluding DML and QUERY,
+            verify DDL are logged in audit log correctly. Leaves audit log enabled this way.
+        """
+        if not self.target_node.is_enterprise:
+            raise UnsupportedNemesis("Auditing feature is only supported by Scylla Enterprise")
+
+        if store == "syslog" and self._is_it_on_kubernetes():
+            raise UnsupportedNemesis("syslog store is not supported on Kubernetes #scylla-operator/1299")
+
+        audit = Audit(self.cluster)
+
+        if audit.is_enabled():
+            audit.disable()
+            raise UnsupportedNemesis("Audit was enabled -> disabling it")
+
+        audit_keyspace = "audit_keyspace"
+        keyspaces_for_audit = [audit_keyspace]
+        InfoEvent(f"Enabling full audit for keyspaces: {keyspaces_for_audit}").publish()
+        audit_config = AuditConfiguration(
+            store=store,
+            categories=["DCL", "DDL", "AUTH", "ADMIN", "DML", "QUERY"],
```

Comment:
In case audit is extended with more categories, like `OTHER`, `ERROR` and `PREPARE`, which Cassandra offers, this test won't be testing all of the categories anymore. I wonder if there's any way to always be sure we enable all categories of statements. Perhaps we could parse scylla's config parameters (either `scylla.yaml` or `--help` output) to see what categories are available, but I believe not all of them are exposed properly, we may miss or even hide some in the future, so this is also unreliable.

Reply:
I think the only way is to stay tuned to audit changes and adapt accordingly - even changing nemesis for the new cases. E.g. currently here we verify only DML, QUERY and DDL. We can think of extending it also to other categories - but this is out of the scope now.

Reply:
IMO we could parse scylla.yaml, because I believe it's dev's responsibility to keep it up-to-date, but I guess we can also just change the nemesis if that happens.

Reply:
We'll adapt once those would be available, as we do for any new feature / bug fixes. i.e. once we'll fix/add the dtest for this feature, we'll need to fix/add those to SCT as well.